File: src/test/dk/ilios/asciihexgrid/CharGridTest.java
package dk.ilios.asciihexgrid;

import junit.framework.TestCase;

public class CharGridTest extends TestCase {

    public void testEmptyGrid() {
        CharGrid grid = new CharGrid(2,4);
        assertEquals("  \n  \n  \n  \n", grid.print(false));
    }

    public void testInput_String() {
        CharGrid grid = new CharGrid(2,2);
        grid.addString(0,0, "12");

        assertEquals("12\n  \n", grid.print(false));
    }

    public void testInput_MultipleLines() {
        CharGrid grid = new CharGrid(2,2);
        grid.addString(0,0, "12");
        grid.addString(0,1, "34");

        assertEquals("12\n34\n", grid.print(false));
    }

    public void testToLongInput() {
        CharGrid grid = new CharGrid(2,2);

        try {
            grid.addString(0,0, "123");
            fail("No exception");
        } catch (IndexOutOfBoundsException e) {
            assertTrue(true);
        }
    }
}

File: src/test/kotlin/dk/ilios/asciihexgrid/CharGridTest.kt
package dk.ilios.asciihexgrid

import junit.framework.TestCase

class CharGridTest : TestCase() {
    fun testEmptyGrid() {
        val grid = CharGrid(2, 4)
        assertEquals("  \n  \n  \n  \n", grid.print(false))
    }

    fun testInput_String() {
        val grid = CharGrid(2, 2)
        grid.addString(0, 0, "12")
        assertEquals("12\n  \n", grid.print(false))
    }

    fun testInput_MultipleLines() {
        val grid = CharGrid(2, 2)
        grid.addString(0, 0, "12")
        grid.addString(0, 1, "34")
        assertEquals("12\n34\n", grid.print(false))
    }

    fun testToLongInput() {
        val grid = CharGrid(2, 2)
        try {
            grid.addString(0, 0, "123")
            fail("No exception")
        } catch (e: IndexOutOfBoundsException) {
            assertTrue(true)
        }
    }
}